### PR TITLE
keys source build

### LINKED ISSUE #3246: extend build from source to KEYS

### PR DESCRIPTION
**Pull request by gearama**
closes #3246 
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Comment by vhvb1989** [comment]
Fix conflicts and use latest main

**Comment by gearama** [comment]
the core versions will be decided in the next PR

File: sdk/keyvault/azure-security-keyvault-keys/CMakeLists.txt [hunk 1]
```diff
@@ -118,7 +135,7 @@ generate_documentation(azure-security-keyvault-keys ${AZ_LIBRARY_VERSION})
 
 if(BUILD_TESTING)
 
-  if (NOT AZ_ALL_LIBRARIES)
+  if (NOT AZ_ALL_LIBRARIES OR FETCH_SOURCE_DEPS)
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-keys/CMakeLists.txt:138]: <3

**Review by vhvb1989 — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-keys/CMakeLists.txt (hunk 1 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-keys/CMakeLists.txt:138]: :D

**Pull request merged**